app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts

```diff
@@ -567,6 +571,19 @@ export class AwsServerGroupConfigurationService {
         return result;
       };
 
+      command.clusterChanged = (): void => {
+        // we stringify here because the model (IServerGroupCommand) expects an Application object
+        // but at runtime a string is given.  So to avoid a compilation error, we stringify
+        const appName = JSON.stringify(command.application);
```

Comment:
We should really fix this upstream so we have the full application object (and not just the name), as the model expects.

I think these are the only two lines that need to change:
* https://github.com/spinnaker/deck/blob/master/app/scripts/modules/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js#L48
* https://github.com/spinnaker/deck/blob/master/app/scripts/modules/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js#L192

I tested this locally, changing `application.name` to `application`, and everything seemed fine.

Reply:
@anotherchrisberry - I've made these changes and you should see them reflected in the new commit.